url647b55da7c_Nikhilkumar13_emuSr_tgz-pJ8-MainJ8/src/frontend/handlers.java

```diff
@@ -0,0 +1,959 @@
+                }
+            }
+           // FrontEnd.statuswindow.append("PROGRAM STARTED " + FrontEnd.filepath + "\n");
+            FrontEnd.appendToPane(FrontEnd.statuswindow,"PROGRAM STARTED " + FrontEnd.filepath + "\n",Color.BLACK);
+            FrontEnd.appendToPane(FrontEnd.statuswindow,"Running..\n",Color.GREEN);
+
+            
+            //frontend.FrontEnd.statuswindow.setCaretPosition(frontend.FrontEnd.statuswindow.getText().length());
+            clean_branchtable();
+
+            clean_memtable();
+            Integer ct = new Integer(0);
+            ct = backend.FirstPass.scan(fin);
+            //System.out.println("ct returned:" + ct);
+//            FrontEnd.statuswindow.append("Running.. \n");
+            //FrontEnd.appendToPane(FrontEnd.statuswindow,"Runing...\n",Color.BLACK);
+
+            
+            //frontend.FrontEnd.statuswindow.setCaretPosition(frontend.FrontEnd.statuswindow.getText().length());
+            try
+            {
+                backend.ScanFile ob = new backend.ScanFile(fin);
+                update_branchtable(ct);
+                update(handlers.regMode);
+
+            } catch (Exception e)
+            {
+            }
+        } else
+        {
+            try
+            {
+                // Create temp file.
+                File temp = File.createTempFile("temp", ".s");
+
+                FrontEnd.filepath = null;
+                out = null;
+                try
+                {
+
+                    out = new BufferedWriter(new FileWriter(temp));
+                    out.write(FrontEnd.activepane.getText());
+                } catch (IOException ex)
+                {
+                    //FrontEnd.statuswindow.append("Error occured ...\n");
+                    FrontEnd.appendToPane(FrontEnd.statuswindow,"Error Occured \n",Color.BLACK);
+
+//                    FrontEnd.statuswindow.append(ex.getStackTrace().toString());
+                     FrontEnd.appendToPane(FrontEnd.statuswindow,ex.getStackTrace().toString(),Color.BLACK);
+                    //frontend.FrontEnd.statuswindow.setCaretPosition(frontend.FrontEnd.statuswindow.getText().length());
+                } finally
+                {
+                    try
+                    {
+                        out.close();
+                    } catch (IOException ex)
+                    {
+//                        FrontEnd.statuswindow.append("Error occured ...\n");
+//                        FrontEnd.statuswindow.append(ex.getStackTrace().toString());
+                         
+                        FrontEnd.appendToPane(FrontEnd.statuswindow,"Error Occured...\n",Color.BLACK);
+                         FrontEnd.appendToPane(FrontEnd.statuswindow,ex.getStackTrace().toString(),Color.BLACK);
+                        //frontend.FrontEnd.statuswindow.setCaretPosition(frontend.FrontEnd.statuswindow.getText().length());
+                        //frontend.FrontEnd.exceptionraised++;
+                    }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 358:370, 578:589.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.